### ISSUE TITLE
Paypal IPN scripts to use HTTP 1.1

### ISSUE THREAD

**Issue by Actros81** [dioscouri/tienda#139]
https://www.x.com/node/320404

Action Required before February 1, 2013
Merchants need to update their IPN and/or PDT scripts to use HTTP 1.1, and include the “Host” and “Connection” headers in the IPN postback script.

PHP
// post back to PayPal system to validate
$header .="POST /cgi-bin/webscr HTTP/1.1\r\n";
$header .="Content-Type: application/x-www-form-urlencoded\r\n";
$header .="Host: www.paypal.com\r\n";
$header .="Connection: close\r\n";

**Comment by ChrisFrench** [comment]
Yep we already have sent a few emails about this   but thanks for making the ticket, The paypal pro  plugin is in need of a rewrite, I have already added in the latest API  to the plugin folder but need to do the rewritting

**Comment by Actros81** [comment]
ok, take in care that also Paypal standard proccess needs  HTTP 1.1

**Comment by Actros81** [comment]
Hi,
I tested a lot this topic, and this solution does't works fine...

Paypal suggest to use the following sequence, and it's works fine in my website

```
    $ch = curl_init('https://www.paypal.com/cgi-bin/webscr'); // or https://www.sandbox.paypal.com/cgi-bin/webscr  for sandbox testing
    curl_setopt($ch, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER,1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, $req);
    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 1);
    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 2);
    curl_setopt($ch, CURLOPT_FORBID_REUSE, 1);
    curl_setopt($ch, CURLOPT_HTTPHEADER, array('Connection: Close'));

    if( !($res = curl_exec($ch)) ) {
        // error_log("Got " . curl_error($ch) . " when processing IPN data");
        curl_close($ch);
        exit;
    }
    curl_close($ch);


    if (strcmp ($res, "VERIFIED") == 0){
    return '';
    } else if (strcmp ($res, "INVALID") == 0) {
    return JText::_('PAYPAL ERROR IPN VALIDATION');
    }
```

**Comment by ChrisFrench** [comment]
OK, I just changed the header  I will spend more time on this and test your solution

**Comment by ChrisFrench** [comment]
they fopensock is working just fine on my local system, I added an option for CURL  to the config and we can have the option to support  as many servers as possible. 

Can you check this, I am going to leave this open  until  I have time to test more

**Comment by Actros81** [comment]
Yey, sure. Let me test this solution

**Comment by Kathile** [comment]
Could you let me know if this is a problem for folks running the 0.9 version of Tienda?
